ambry-commons/src/main/java/com/github/ambry/compression/ZstdCompression.java

```diff
@@ -0,0 +1,137 @@
+/**
+ * Copyright 2022 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.compression;
+
+import com.github.luben.zstd.Zstd;
+
+/**
+ * Zstandard compression algorithm using Zstandard native and JNI.
+ * Zstandard has fast decompression speed and pretty good compression ratio.
+ * It is not as fast as LZ4 but has higher compression ratio than LZ4.
+ * More info about Zstandard in <a href="https://github.com/lz4/lz4">GitHub</a>.
+ *
+ * Zstd compression level range is from negative 7 (fastest) to 22 (slowest in compression speed,
+ * but best compression ratio), with level 3 as the default.
+ */
+public class ZstdCompression extends BaseCompressionWithLevel {
+
+  /**
+   * Get the unique name of this compression algorithm.
+   * WARNING - Do not change the algorithm name.  See Compression interface for detail.
+   * @return Name of this algorithm.
+   */
+  @Override
+  public String getAlgorithmName() {
+    return "ZSTD";
```

Comment:
public static final string/int for constants?

Reply:
Also maybe use enums for supported algorithm names.

Reply:
No enum for this case. Making it enum means adding a new algorithm will need to modify the enum.  Say, another team wants to use our library and they want to provide their own compression algorithm, they can't modify our enum class to add their class.  Enum is useful to make code readable but limits expansion